packages/kbn-config/tsconfig.json

```diff
@@ -0,0 +1,21 @@
+{
+  "extends": "../../tsconfig.base.json",
+  "compilerOptions": {
+    "declaration": true,
+    "declarationDir": "./target/types",
+    "outDir": "./target/out",
+    "stripInternal": false,
+    "declarationMap": true,
+    "types": [
+      "jest",
+      "node"
+    ]
+  },
+  "include": [
+    "./src/**/*.ts",
+    "../../typings/index.d.ts"
```

Comment:
JFYI: I extracted `PublicMethodsOf` in `@kbn/utility-types` https://github.com/elastic/kibana/pull/76785